server/src/main/java/com/linecorp/centraldogma/server/metadata/ProjectRoles.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2024 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.centraldogma.server.metadata;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.MoreObjects;
+
+import com.linecorp.armeria.common.annotation.Nullable;
+import com.linecorp.centraldogma.common.RepositoryRole;
+
+/**
+ * Represents the roles assigned to project members and guests for a specific repository.
+ */
+public final class ProjectRoles {
+
+    /**
+     * Returns a new {@link ProjectRoles} with the specified {@link RepositoryRole}s.
+     */
+    public static ProjectRoles of(@Nullable RepositoryRole member, @Nullable RepositoryRole guest) {
+        return new ProjectRoles(member, guest);
```

Comment:
nit: Create `EMTPY_PROJECT_ROLES` and reuse if both `member` and `guest` are null?